modules/swagger-codegen/src/main/resources/asyncscala/api.mustache

```diff
@@ -40,9 +40,14 @@ class {{classname}}(client: TransportClient, config: SwaggerConfig) extends ApiC
     {{/required}}
     {{/queryParams}}
 
-    {{#headerParams}}
-    headerParams += "{{baseName}}" -> {{paramName}}.toString
-    {{/headerParams}}
+  {{#headerParams}}
+      {{^required}}
+          if ({{paramName}} != null) {{paramName}}.map { v => headerParams += "{{baseName}}" -> v.toString }
+      {{/required}}
+      {{#required}}
+          if ({{paramName}} != null) headerParams += "{{baseName}}" -> {{paramName}}.toString
+      {{/required}}
```

Comment:
Ah I see. But why do we use Option for non-required and raw type for required params? Required or not can be achieved by whether providing default value or not. It doesn't matter whether we use Option or not. It's orthogonal.